packages/config/src/Config.ts

```diff
@@ -0,0 +1,666 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import * as node_path from "path";
+import * as fs from "fs";
+import * as deepmerge from "deepmerge";
+import * as findUp from "find-up";
+import * as JSONC from "comment-json";
+import * as lodash from "lodash";
+
+import { IConfig } from "./doc/IConfig";
+import { IConfigLayer } from "./doc/IConfigLayer";
+import { ImperativeError } from "../../error";
+import { IConfigProfile } from "./doc/IConfigProfile";
+import { IConfigOpts } from "./doc/IConfigOpts";
+import { IConfigSecure, IConfigSecureProperties } from "./doc/IConfigSecure";
+import { IConfigVault } from "./doc/IConfigVault";
+import { Logger } from "../../logger";
+import { IConfigLoadedProfile, IConfigLoadedProperty } from "./doc/IConfigLoadedProfile";
+
+enum layers {
+    project_user = 0,
+    project_config,
+    global_user,
+    global_config
+};
+
+export class Config {
+    private static readonly SECURE_ACCT = "secure_config_props";
+
+    private _app: string;
+    private _paths: string[];
+    private _layers: IConfigLayer[];
+    private _home: string;
+    private _name: string;
+    private _user: string;
+    private _schema: string;
+    private _active: {
+        user: boolean;
+        global: boolean
+    };
+    private _vault: IConfigVault;
+    private _secure: IConfigSecure;
+
+    private constructor() { }
+
+    public static readonly INDENT: number = 4;
+
+    public static empty(): IConfig {
+        return {
+            profiles: {},
+            defaults: {},
+            plugins: [],
+            secure: []
+        };
+    }
+
+    public static async load(app: string, opts?: IConfigOpts): Promise<Config> {
+        opts = opts || {};
+
+        ////////////////////////////////////////////////////////////////////////
+        // Create the basic empty configuration
+        const _ = new Config();
+        (_ as any).config = {};
+        _._layers = [];
+        _._home = opts.homeDir || node_path.join(require("os").homedir(), `.${app}`);
```

Comment:
should `os` be imported similarly to `import * as fs from "fs";` above to be used here and line 86?